integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/SearchQueryBaseIT.java

```diff
@@ -77,27 +80,31 @@ public class SearchQueryBaseIT {
 	private static final String TITLE_AVENUE_OF_MYSTERIES = "Avenue of Mysteries";
 	private static final String AUTHOR_AVENUE_OF_MYSTERIES = "John Irving";
 
-	@ClassRule
-	public static BackendMockTestRule backendMock = BackendMockTestRule.createGlobal();
+	@RegisterExtension
+	public static BackendMock backendMock = BackendMock.create();
 
-	@ClassRule
-	public static ReusableOrmSetupHolder setupHolder = ReusableOrmSetupHolder.withBackendMock( backendMock );
+	@RegisterExtension
+	public static OrmSetupHelper ormSetupHelper = OrmSetupHelper.withBackendMock( backendMock );
+	private SessionFactory sessionFactory;
 
-	@Rule
-	public MethodRule setupHolderMethodRule = setupHolder.methodRule();
+	@DataClearConfigConfigurer
+	public void tearDown(DataClearConfigConfigurer.DataClearConfig config) throws Exception {
+		config.clearOrder( Book.class, Author.class, NotIndexed.class );
+	}
```

Comment:
Hey Yoann,

The idea here is to move the cleaning-up logic into the `OrmSetupHelper` itself. If there's a `@DataClearConfigConfigurer` annotated method in the test class using this helper then the helper will clean the data after the test execution. It doesn't look like we need to care about tracking the configuration parameters anymore. Since we have a `@ParameterizedClass` it should take care of setting up the session factory, while the only thing remaining is the data cleanup.

From what it looks, we don't really need to have the `ReusableOrmSetupHolder` in all the cases... (see previous commit where I've updated a few tests previously relying on this holder). Or we don't need to cleanup the data between tests in some examples as well (wich may help us speed up the execution a bit).
Also, we don't need the `@ParameterizedClass` in all cases since we can easily convert the test to use `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` + `@BeforeAll` and simple `OrmSetupHelper` + adding the `@DataClearConfigConfigurer` when we need to clean the data.

What do you think of this approach?

Reply:
Hey,

Good points.

I don't think we need an annotations or a dedicated method in the test, though, do we?

You could just allow something like that:

```
setupHelper.start(...)
    .dataClearing(config -> { ... })
    .setup();
```

The call to `clearAfterTest` would store some runnable in the extension or test context, and the runnable would get executed after each test.

Also... I think we probably want data clearing to get executed by default (opt-out), at least when the session factory is set up in `@BeforeAll`. See my other comments for the "why".

Reply:
That was a nice idea, I went with the annotations as that was how we were doing it before, but defining it in the helper makes more sense 😃 👍🏻. I think I've managed to make it delete the DB data if the helper is static, as for the index -- from what it seems, it's better to opt-in for that since a lot of our tests are based on the mock and if we drop-create the index -- it'll generate an additional "event" that we need to add to the expectations. + this wasn't happening in the past so that seems like a safe default ...

Reply:
> this wasn't happening in the past

I beg to differ:

https://github.com/hibernate/hibernate-search/blob/5e8810801f9c920c5d59fd7ff201f91371673053/util/internal/integrationtest/mapper/orm/src/main/java/org/hibernate/search/util/impl/integrationtest/mapper/orm/ReusableOrmSetupHolder.java#L394-L402

> as for the index -- from what it seems, it's better to opt-in for that since a lot of our tests are based on the mock and if we drop-create the index

You could also just adjust the behavior based on whether the backend is a mock or not; that's precisely what we used to do in `ReusableOrmSetupHolder`:

https://github.com/hibernate/hibernate-search/blob/5e8810801f9c920c5d59fd7ff201f91371673053/util/internal/integrationtest/mapper/orm/src/main/java/org/hibernate/search/util/impl/integrationtest/mapper/orm/ReusableOrmSetupHolder.java#L155-L172

You could change how it works, e.g. add something to `BackendSetupStrategy`, if that's easier. But my point is you can avoid the index clearing when using a backend mock :)

Reply:
riiiiiiight 😃 
When I was saying `this wasn't happening in the past` I meant in the regular `OrmSetupHelper`. I've took the bits from the reusable one so I've seen the part you are pointing to. It was done with `ReusableOrmSetupHolder#withSingleBackend` so Ive added an explicit config to clear the index in these cases... I am just wondering if it is worth to clear the index if it wasn't cleared before (in terms of time-to-execute-the-tests). If you think it won't impact the time that much and it is worth I'll figure something 😃

Reply:
> am just wondering if it is worth to clear the index if it wasn't cleared before (in terms of time-to-execute-the-tests).

IMO if we clear the database we must clear the indexes as well (except for mock backends, obviously). Otherwise we'll run into problems.

Is clearing the indexes a big cost? Not that much, especially considering that the vast majority of ORM tests use a `BackendMock`, where we won't (shouldn't) clear the indexes anyway. For other tests, we might lose 50ms per clear; we'll survive.

Now, is it worth clearing the indexes all the time? I'd say no, it's only worth it if we clear the database, and that should only happen when the `SessionFactory` was created in a `@BeforeAll`... and I think you already implemented that optimization, so it should be fine :)